### ISSUE TITLE
skim mode slow on initial search

### ISSUE THREAD

**Issue by tribly** [atuinsh/atuin#958]
I've been trying atuin and am happy to see that there's a skim search mode!

Unfortunately the initial search (after the first keypress) when interactive takes about 0.3s to get some results and update the UI, blocking the UI which feels incredibly laggy. After that, it's super snappy.

OS: Arch
Version: 14.0.1-1

**Comment by conradludgate** [comment]
Yeah, we haven't optimised skim search yet. We're not sure whether it's gonna stay or not. It currently loads the entire database on first search, then the in memory search is fast after that

**Comment by ristomatti** [comment]
Without knowing Rust or the current implementation, perhaps some of these optimizations would be relatively low effort to implement while still making a noticeable improvement?

1. If this is not the current behavior already, only load commands matching the active filter mode in memory. I've set Control+R to filter by host (42500 entries) and Up key by session (28 entries). The initial keystroke lag feels to be equally slow.
2. Start loading the database into memory right after the search interface has loaded. This would reduce the perceived lag even if it's not actually improving the performance.
3. Load the database into memory in a non-blocking way. Currently you'll have to wait for it to complete before the first characters appear. This should make the search feel snappier. Possibly it could even make the subsequent search faster by being more specific with the characters typed while the data is loading.

**Comment by dotfrag** [comment]
Also hoping for this to be resolved. At least for my use case, the bias towards consecutive matches from skim is superior to other fuzzy algorithms. The default fuzzy algorithm doesn't return the intended item most of time. Solution no. 2 by @ristomatti is definitely acceptable.

**Comment by 0xC0FFEE** [comment]
Yes, the irony is that I came from FZF, and Atuin advertises with: 

> Search decades of shell history and recall it in an instant.

Despite my decades of shell history, fzf was always blazingly fast. Now, using Atain with Skim (the other mode returns low-quality results) is slower than fzf :(

Can we help in making this fast(er)?

**Comment by ellie** [comment]
just an update for you all! We haven't fixed skim, because the search performance there is due to a fundamental difference in how thing work.

with the release of `18.13.2`, we now ship a new search mode. This is _much_ more like the search you'd get with `fzf`, as it uses [nucleo](https://github.com/helix-editor/nucleo) behind the scenes. It requires that you use the Atuin Daemon - it keeps a search index warm and in-memory, so that it doesn't require building from scratch every time + should be very fast. We also have a bunch of frecency scoring changes, that should feel much better.

To enable, you have two options

1. Run `atuin setup`, say yes to the daemon. Good to go.
2. Edit your `config.toml`, with the following changes
```toml
search_mode = "daemon-fuzzy"

[daemon]
enabled = true
autostart = true
```

**Comment by ellie** [comment]
Gonna close this one for now, happy to reopen if ^ does not work for you. We will likely be deprecating skim in future releases, in preference of the new daemon search.
